manager/secret_manager.go

```diff
@@ -8,6 +8,30 @@ func (r *ResourceManager) CreateSecret(namespace string, secret *model.Secret) (
 	return r.kubeClient.CreateSecret(namespace, secret)
 }
 
-func (r *ResourceManager) GetSecret(namespace, name string) (secret *model.Secret, err error) {
-	return r.kubeClient.GetSecret(namespace, name)
+func (r *ResourceManager) SecretExists(namespace string, secretName string) (exists bool, err error) {
+	return r.kubeClient.SecretExists(namespace, secretName)
+}
+
+func (r *ResourceManager) GetSecret(namespace string, secret *model.Secret) (secretRes *model.Secret, err error) {
+	return r.kubeClient.GetSecret(namespace, secret.Name)
+}
+
+func (r *ResourceManager) ListSecrets(namespace string) (secrets []model.Secret, err error) {
```

Comment:
The return var should be `secrets []*model.Secret`

Reply:
I'll do this change, but why use []*?

Reply:
Good question, it's possible these secrets can go up to 1MB (let's say if there is a cert or multiple certs in them). In this case you're not copying each struct if you iterate through the slice or pass the slice from method to another.

Reply:
Interesting, good to know.